kernel/filesystem/Filesystem.cpp

```diff
@@ -301,3 +316,21 @@ Result filesystem_rename(Path *old_path, Path *new_path)
 
     return result;
 }
+
+Result filesystem_chmod(Path *path, uint mode)
+{
+    auto parent = filesystem_find_parent_and_ref(path);
+
+    if (!parent || !path_filename(path))
+    {
+        return ERR_NO_SUCH_FILE_OR_DIRECTORY;
+    }
+
+    Mode fmode = {(mode | MODE_USER_READ) == MODE_USER_READ, (mode | MODE_USER_WRITE) == MODE_USER_WRITE, (mode | MODE_USER_EXECUTE) == MODE_USER_EXECUTE, (mode | MODE_GROUP_READ) == MODE_GROUP_READ, (mode | MODE_GROUP_WRITE) == MODE_GROUP_WRITE, (mode | MODE_GROUP_EXECUTE) == MODE_GROUP_EXECUTE, (mode | MODE_OTHER_READ) == MODE_OTHER_READ, (mode | MODE_OTHER_WRITE) == MODE_OTHER_WRITE, (mode | MODE_OTHER_EXECUTE) == MODE_OTHER_EXECUTE};
```

Comment:
This line is way too long 😅 maybe you can pass the mode structures directly to prevent that

Reply:
Done